prow/cmd/splice/main.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+var (
+	submitQueueURL = flag.String("submit-queue-endpoint", "http://submit-queue.k8s.io/github-e2e-queue", "Submit Queue status URL")
+	remoteURL      = flag.String("remote-url", "https://github.com/kubernetes/kubernetes", "Remote Git URL")
+	repoName       = flag.String("repo", "kubernetes/kubernetes", "Repo name")
+	logJson        = flag.Bool("logjson", false, "output log in JSON format")
+)
+
+// Call a binary and return its output and success status.
+func call(binary string, args ...string) (string, error) {
+	cmdout := "+ " + binary + " "
+	for _, arg := range args {
+		cmdout += arg + " "
+	}
+	log.Debug(cmdout)
+
+	cmd := exec.Command(binary, args...)
+	output, err := cmd.CombinedOutput()
+	return string(output), err
+}
+
+// getQueuedPRs reads the list of queued PRs from the Submit Queue.
+func getQueuedPRs(url string) ([]int, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+
+	queue := struct {
+		E2EQueue []struct {
+			Number int
+		}
+	}{}
+	err = json.Unmarshal(body, &queue)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := []int{}
+	for _, e := range queue.E2EQueue {
+		ret = append(ret, e.Number)
+	}
+	return ret, nil
+}
+
+// Splicer manages a git repo in specific directory.
+type splicer struct {
+	dir string // The repository location.
+}
+
+// makeSplicer returns a splicer in a new temporary directory,
+// with an initial .git dir created.
+func makeSplicer() (*splicer, error) {
```

Comment:
it's kinda gross that if this returns error then you may or may not have left a temp dir hanging around.